src/axom/mir/detail/elvira_detail.hpp

```diff
@@ -346,6 +407,416 @@ class TopologyBuilder<ExecSpace, CoordsetView, TopologyView, MatsetView, Polygon
+   * \param[out] n_coordset The Conduit node in which to build the new coordset.
+   * \param[out] n_topology The Conduit node in which to build the new topology.
+   * \param[out] n_fields The Conduit node in which to build the new fields.
+   * \param[out] n_matset The Conduit node in which to build the new matset.
+   * \param n_options A Conduit node that contains MIR options.
+   *
+   * \note This is a host-only function. We allocate fixed size arrays for the coordset and
+   *       connectivity that will have gaps that consumers will need to skip over using
+   *       the provided sizes/offsets.
+   */
+  void allocate(axom::IndexType numFragments,
+                axom::IndexType maxCuts,
+                conduit::Node &n_coordset,
+                conduit::Node &n_topology,
+                conduit::Node &n_fields,
+                conduit::Node &n_matset,
+                const conduit::Node &n_options)
+  {
+    namespace bputils = axom::mir::utilities::blueprint;
+    bputils::ConduitAllocateThroughAxom<ExecSpace> c2a;
+
+    // Handle options
+    const std::string originalElementsField(MIROptions(n_options).originalElementsField());
+    if(n_options.has_child("normal"))
+    {
+      m_view.m_makeNormal = (n_options.fetch_existing("normal").to_int() != 0);
+    }
+
+    // Figure out some fragment size information given maxCuts, the max number of
+    // times a zone will be cut.
+    m_view.m_MAX_POINTS_PER_FACE = 6 + maxCuts;
+    m_view.m_MAX_FACES_PER_FRAGMENT = 6 + maxCuts;
+    m_view.m_MAX_POINTS_PER_FRAGMENT = 8 + maxCuts * 2;
+
+    // Set up coordset and allocate data arrays.
+    // Note that we overallocate the number of nodes to numCoordValues.
+    const auto numCoordValues = numFragments * m_view.m_MAX_POINTS_PER_FRAGMENT;
+    n_coordset["type"] = "explicit";
+    n_coordset["values/x"].set_allocator(c2a.getConduitAllocatorID());
+    n_coordset["values/x"].set(conduit::DataType(bputils::cpp2conduit<CoordType>::id, numCoordValues));
+    m_view.m_x = bputils::make_array_view<CoordType>(n_coordset["values/x"]);
+    n_coordset["values/y"].set_allocator(c2a.getConduitAllocatorID());
+    n_coordset["values/y"].set(conduit::DataType(bputils::cpp2conduit<CoordType>::id, numCoordValues));
+    m_view.m_y = bputils::make_array_view<CoordType>(n_coordset["values/y"]);
+    n_coordset["values/z"].set_allocator(c2a.getConduitAllocatorID());
+    n_coordset["values/z"].set(conduit::DataType(bputils::cpp2conduit<CoordType>::id, numCoordValues));
+    m_view.m_z = bputils::make_array_view<CoordType>(n_coordset["values/z"]);
+
+    axom::mir::utilities::fill<ExecSpace>(m_view.m_x, CoordType(0));
+    axom::mir::utilities::fill<ExecSpace>(m_view.m_y, CoordType(0));
+    axom::mir::utilities::fill<ExecSpace>(m_view.m_z, CoordType(0));
+
+    // elements (zone definitions)
+    constexpr ConnectivityType UnusedValue =
+      std::numeric_limits<ConnectivityType>::is_signed ? -1 : 99999999;
```

Comment:
Should we use `std::numeric_limits<ConnectivityType>::max()` instead of a hardcoded `9999999`?

Reply:
@publixsubfan - I did that in one of the follow-up PRs.